### PR TITLE
refactor: replace OrderedDict with dict

### LINKED ISSUE #369: Fix types when migrating to Python 3.9
Upgrading to Python 3.9 allows us to fix a lot of types. In particular:

- Remove `OrderedDict`
- Explicit the correct types for classes that inherit from the `Mapping` abc (as `Collections`)
- Use `list` instead of `List`
- etc

### PR DESCRIPTION
**Pull request by sergkudinov**
#### Which issue(s) this PR fixes

Part of #369

#### Additional comments

This seams to be already fixed:

> - Explicit the correct types for classes that inherit from the `Mapping` abc (as `Collections`)
> - Use `list` instead of `List`

I am not sure if there is smth else to refactor. Checked with [whats new in python 3.9](https://docs.python.org/3/whatsnew/3.9.html).

#### Agreements

- [x] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0).
- [x] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

**Comment by rpignolet** [comment]
I don't know why you want to use `dict` instead of `OrderedDict`.

**Comment by sergkudinov** [comment]
In the issue you mention: 

> Remove `OrderedDict`

I understood we replace `OrderedDict`, which `dict` does the same since Python 3.6, it also keeps items ordered ([reference](https://realpython.com/python-ordereddict/#choosing-between-ordereddict-and-dict)) and a bit more efficiente than `OrderedDict`.

To my mind, it worth to do that.

**Comment by sergkudinov** [comment]
@PaulFarault @rpignolet 
what do you think?

**Comment by rpignolet** [comment]
I think is better to keep `OrderedDict` to no rely on detail implementation. Moreover, with `OrderedDict` it's explicit that we want an order for keys.

**Comment by PaulFarault** [comment]
Thx. As we're looking at it, could you add a commit to fix the Iterator please (make it a Generator).

**Review by rpignolet — APPROVED** [review]

File: tdp/core/variables/service_variables.py [hunk 1]
```diff
@@ -180,9 +180,7 @@ def _open_var_file(
     @contextmanager
     def open_var_files(
         self, message: str, paths: list[str], fail_if_does_not_exist: bool = False
-    ) -> Iterator[
-        dict[str, Variables]
-    ]:  # TODO: Transform Dict to OrderedDict with python>3.6
+    ) -> Generator[OrderedDict[str, Variables]]:
```
Comment (PaulFarault) [inline tdp/core/variables/service_variables.py:183]: Generator generic takes 3 type parameters: `Generator[YieldType, SendType, ReturnType]`.

```suggestion
    ) -> Generator[OrderedDict[str, Variables], None, None]:
```

**Review by PaulFarault — CHANGES_REQUESTED** [review]
I saw that you linked this PR to #369 which will close it. However, it seems that there are still some types to migrate. Could you please either fix the whole issue or transform the `Fixes` keyword to `Part of`?

File: tdp/core/variables/service_variables.py (hunk 1 above)
Comment (sergkudinov) [reply tdp/core/variables/service_variables.py]: done

**Review by PaulFarault — APPROVED** [review]

**Pull request merged**